src/parameterizations/vertical/MOM_diabatic_driver.F90

```diff
@@ -2588,10 +2586,11 @@ subroutine extract_diabatic_member(CS, opacity_CSp, optics_CSp, evap_CFL_limit,
                                                             !! assume thermodynamics properties are valid.
 
   ! Pointers to control structures
-  if (present(opacity_CSp))       opacity_CSp => CS%opacity_CSp
+  if (present(opacity_CSp))       opacity_CSp => CS%opacity
   if (present(optics_CSp))        optics_CSp  => CS%optics
   if (present(KPP_CSp))           KPP_CSp     => CS%KPP_CSp
-  if (present(energetic_PBL_CSp)) energetic_PBL_CSp => CS%energetic_PBL_CSp
+  if (present(energetic_PBL_CSp)) energetic_PBL_CSp => CS%energetic_PBL
```

Comment:
With the NCAR configuration, we are getting a memory access error due to the above line (2592).

I should say I can't fully comprehend the data ownership and flow regarding the energetic_PBL module and its control structure, but here is summary of my understanding:
- Regardless of the `ENERGETICS_SFC_PBL` runtime parameter, `CS%energetic_PBL` is always allocated when `diabatic_CS` gets allocated since `energetic_PBL` is not a pointer (or a target), but rather a data object within `diabatic_CS`.
- The issue is that the automatic allocation of the `CS%energetic_PBL` control structure doesn't guarantee the runtime initialization of the `CS%energetic_PBL` data members via the `energetic_PBL_init` subroutine call.
- In other parts of the code, such as MOM_neutral_diffusion.F90:L326, it is incorrectly assumed that if `energetic_PBL_CSp` is associated, then `energetic_PBL_init` must have been called, which is not always the case.
- While I am not sure what the proper solution should look like, the following diff for the above line (2592) fixed the issue for me:

```
-  if (present(energetic_PBL_CSp)) energetic_PBL_CSp => CS%energetic_PBL
+  if (present(energetic_PBL_CSp) .and. CS%use_energetic_PBL) energetic_PBL_CSp => CS%energetic_PBL
```

Reply:
Thank you @alperaltuntas, I agree your explanation and with this fix.  There were many other examples of this sort of issue during removal of the other pointers, and this one must have been overlooked.

I also reckon this variable should be renamed to `ePBL` to avoid confusion with the `energetic_PBL` subroutine.

Reply:
Fix included in this branch:https://github.com/marshallward/mom6/tree/check_use_epbl

Reply:
Thanks, @marshallward! I just confirmed that merging your branch works for us. So this issue can be marked resolved when your branch gets merged. I am now running the remainder of our test suite.

Reply:
It's also passed our local regression, so I've updated the PR with your patch.  Thanks again!